luigi/static/visualiser/js/graph.js

```diff
@@ -58,21 +58,35 @@ Graph = (function() {
     /* Compute the maximum depth of each node for layout purposes, returns the number
        of nodes at each depth level (for layout purposes) */
     function computeDepth(nodes, nodeIndex) {
-        var rowSizes = [];
         function descend(n, depth) {
-            n.depth = depth;
+            if (n.depth === undefined || depth > n.depth) {
```

Comment:
So is the slow user experience because of this? I assumed it was slow because of rendering...